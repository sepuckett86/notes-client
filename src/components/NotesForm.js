import React from 'react';
import PropTypes from 'prop-types';

function NotesForm({ onChange, onSubmit, title, body }) {
  return (
    <form onSubmit={onSubmit}>
      <label>
        Title:{' '}
        <input name="title" type="text" value={title} onChange={onChange}/>
      </label>
      <label>
        Body:{' '}
        <input name="body" type="text" value={body} onChange={onChange}/>
      </label>
      <button>Submit</button>
    </form>
  );
}

NotesForm.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired
};

export default NotesForm;
