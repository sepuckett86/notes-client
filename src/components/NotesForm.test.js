import React from 'react';
import { shallow } from 'enzyme';
import NotesForm from './NotesForm';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<NotesForm 
      title="My Title"
      body="My really good note"
      onChange={() => {}}
      onSubmit={() => {}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
