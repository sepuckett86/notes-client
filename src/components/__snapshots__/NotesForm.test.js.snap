// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Title:
     
    <input
      name="title"
      onChange={[Function]}
      type="text"
      value="My Title"
    />
  </label>
  <label>
    Body:
     
    <input
      name="body"
      onChange={[Function]}
      type="text"
      value="My really good note"
    />
  </label>
  <button>
    Submit
  </button>
</form>
`;
